packages/medusa/src/loaders/__tests__/overrides.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { MedusaError } from "medusa-core-utils"
+import { IsString } from "class-validator"
+
+import overridesLoader from "../overrides"
+import { registerOverride, resetOverride } from '../../utils'
+import { validator } from '../../utils/validator'
+import { AdminPostProductsReq } from "../../api/routes/admin/products/create-product"
+
+class ExtendedAdminPostProductsReq extends AdminPostProductsReq {
+  @IsString()
+  custom_attribute: string
+}
+
+async function registrationHelper({ key, override }): Promise<MedusaError | null> {
+  let error = null
+
+  try {
+    registerOverride({
+      key,
+      override
+    })
+
+    await overridesLoader({ isTest: true })
+  } catch (e) {
+    error = e
+  }
+
+  return error
+}
+
+describe("overridesLoader", () => {
+  let error, key, override
+
+  beforeEach(() => {
+    resetOverride()
+  })
+
+  it("should not throw error when registered correctly", async () => {
+    const error = await registrationHelper({
+      key: 'api/routes/store/products/index',
+      override: {
+        defaultStoreProductsFields: ['custom_attribute'],
+        allowedStoreProductsFields: ['custom_attribute'],
+      }
+    })
+
+    expect(error).toBeFalsy()
+  })
+
+  it("should throw error when key does not point to an existing file", async () => {
+    const error = await registrationHelper({
+      key: 'api/incorrect-key',
+      override: {
+        defaultStoreProductsFields: ['custom_attribute'],
+        allowedStoreProductsFields: ['custom_attribute'],
+      }
+    })
+
+    expect(error?.message).toBe(
+      'Override Registration Error: "../api/incorrect-key" not found for registration key "api/incorrect-key"'
+    )
+    expect(error).toBeTruthy()
+  })
+
+  it("should throw error when trying to override a value that does not exist", async () => {
+    const error = await registrationHelper({
+      key: 'api/routes/store/products/index',
+      override: {
+        doesnotexist: ['custom_attribute'],
+      }
+    })
+
+    expect(error?.message).toBe(
+      'Override Registration Error: "doesnotexist" not found in "../api/routes/store/products/index" for registration key "api/routes/store/products/index"'
+    )
+    expect(error).toBeTruthy()
+  })
+
+  it("should correctly register the overriden validator", async () => {
+    let validationResult, error
+
+    validationResult = await validator(AdminPostProductsReq, {
+      title: "test",
+    })
+
+    expect(validationResult.custom_attribute).not.toBeDefined()
```

Comment:
todo: rm, this is not needed as the we are testing the validator and it would have thrown if it was present